contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -167,49 +131,9 @@ contract FunctionsRouter is IFunctionsRouter, FunctionsSubscriptions, Pausable,
   }
 
   // @inheritdoc IRouterBase
-  function proposeConfigUpdateSelf(bytes calldata config) external override onlyOwner {
-    s_proposedConfig[ROUTER_ID] = ConfigProposal({
-      to: config,
-      timelockEndBlock: uint64(block.number + s_timelockBlocks)
-    });
-    emit ConfigProposed({id: ROUTER_ID, toBytes: config});
-  }
-
-  // @inheritdoc IRouterBase
-  function updateConfigSelf() external override onlyOwner {
-    ConfigProposal memory proposal = s_proposedConfig[ROUTER_ID];
-    if (block.number < proposal.timelockEndBlock) {
-      revert TimelockInEffect();
-    }
-    _updateConfig(abi.decode(proposal.to, (Config)));
-    emit ConfigUpdated({id: ROUTER_ID, toBytes: proposal.to});
-  }
-
-  // @inheritdoc IRouterBase
-  function proposeConfigUpdate(bytes32 id, bytes calldata config) external override onlyOwner {
-    s_proposedConfig[id] = ConfigProposal({to: config, timelockEndBlock: uint64(block.number + s_timelockBlocks)});
-    emit ConfigProposed({id: id, toBytes: config});
-  }
-
-  // @inheritdoc IRouterBase
-  function updateConfig(bytes32 id) external override onlyOwner {
-    ConfigProposal memory proposal = s_proposedConfig[id];
-
-    if (block.number < proposal.timelockEndBlock) {
-      revert TimelockInEffect();
-    }
-
-    IConfigurable(getContractById(id)).updateConfig(proposal.to);
-
-    emit ConfigUpdated({id: id, toBytes: proposal.to});
-  }
-
-  // @notice Sets the configuration for FunctionsRouter specific state
-  // @param config bytes of config data to set the following:
-  // - adminFee: fee that will be paid to the Router owner for operating the network
-  function _updateConfig(Config memory config) internal {
+  function updateConfig(Config memory config) public override onlyOwner {
```

Comment:
Maybe a small gas savings to make this calldata instead of memory? It may still need to be loaded into memory, not sure if how smart the struct packing transformation is.

If it does save gas, you could make it calldata in the constructor and other places as well. If the test doesn't work here, assume it won't for the other locations.